// TODO getters/setters
  function testcase() 
  {
    var obj = {
      0 : 11,
      5 : 10,
      10 : 8,
      length : 20
    };
    var accessed = false;
    function callbackfn(val, idx, obj) 
    {
      if (idx > 1)
      {
        accessed = true;
      }
    }
    Object.defineProperty(obj, "1", {
      get : (function () 
      {
        throw new RangeError("unhandle exception happened in getter");
      }),
      configurable : true
    });
    Object.defineProperty(obj, "2", {
      get : (function () 
      {
        accessed = true;
        return 100;
      }),
      configurable : true
    });
    try
{      Array.prototype.forEach.call(obj, callbackfn);
      return false;}
    catch (ex)
{      return (ex instanceof RangeError) && ! accessed;}

  }
  {
    var __result1 = testcase();
    var __expect1 = true;
  }
  
